deployment/terraform/modules/vpc/main.tf

```diff
@@ -0,0 +1,101 @@
+resource "aws_vpc" "vpc" {
+  cidr_block           = "${var.vpc_cidr_block}"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags {
+    Name = "${terraform.workspace} (terraform)"
+  }
+}
+
+resource "aws_subnet" "public_subnet" {
+  vpc_id                  = "${aws_vpc.vpc.id}"
+  cidr_block              = "${var.public_cidr_block}"
+  availability_zone       = "${var.availability_zone_a}"
+  map_public_ip_on_launch = true
+
+  tags {
+    Name = "${terraform.workspace} (terraform public subnet)"
+  }
+}
+
+resource "aws_subnet" "private_subnet_a" {
+  vpc_id            = "${aws_vpc.vpc.id}"
+  cidr_block        = "${var.private_cidr_block_a}"
+  availability_zone = "${var.availability_zone_a}"
+
+  tags {
+    Name = "${terraform.workspace} (terraform private subnet a)"
+  }
+}
+
+/*
+This subnet currently isn't utilized other than to satisfy the
+requirements of module.rds.aws_db_subnet_group. It will be utilized once
+we have a multi-AZ RDS instance
+*/
+resource "aws_subnet" "private_subnet_b" {
+  vpc_id                  = "${aws_vpc.vpc.id}"
+  cidr_block              = "${var.private_cidr_block_b}"
+  availability_zone       = "${var.availability_zone_b}"
+  map_public_ip_on_launch = true
+
+  tags {
+    Name = "${terraform.workspace} (terraform private subnet b)"
+  }
+}
+
+resource "aws_internet_gateway" "public_gateway" {
+  vpc_id = "${aws_vpc.vpc.id}"
+
+  tags {
+    Name = "${terraform.workspace} (terraform)"
+  }
+}
+
+resource "aws_route_table" "route_table" {
+  vpc_id = "${aws_vpc.vpc.id}"
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = "${aws_internet_gateway.public_gateway.id}"
+  }
+
+  tags {
+    Name = "${terraform.workspace} (terraform public)"
+  }
+}
+
+resource "aws_route_table_association" "public_subnet" {
+  subnet_id      = "${aws_subnet.public_subnet.id}"
+  route_table_id = "${aws_route_table.route_table.id}"
+}
+
+############
+
+resource "aws_eip" "nat" {
+  vpc = true
+}
+
+resource "aws_nat_gateway" "nat" {
```

Comment:
Why is a NAT gateway necessary if there is an Internet gateway defined?

Reply:
There might be other ways, is this a blocking issue?

Reply:
This is an unnecessary resource.

Reply:
My sense is that for the private subnets to access the outside world, they do need to go through an NAT: I don't think the regular gateway is sufficient.

I'm less clear if the EIP is necessary.

If we change these, it won't be enough just to make sure apply works: We'll need to bring up a full stack, import visualizations, and run then on this branch, to make sure it all works the same.

Reply:
Yes, NAT gateway is needed for Internet access in private subnets. However, if there is no separate Docker host in this PR then there is no need for the gateway and most certainly no need for the EIP.

Reply:
@hackdna We've realized the docker host portion has been pulled out [here](https://github.com/refinery-platform/refinery-platform/tree/scottx611x%2Fec2_for_docker) and have since removed these resources.